### PR TITLE
sysusers: cross-check user and group names too

### PR DESCRIPTION
**Pull request by lucab**
This adds an additional name check when cross-matching new group entries against existing users, which allows coalescing entries matching both ID and name.
It provides a small idempotence enhancement when creating groups in cases where matching user entries are in place.
By fine-tuning the conflict detection logic, this avoids picking up new random IDs and correctly prefers configuration values instead.

Refs:
 * https://github.com/systemd/systemd/issues/23745#issuecomment-1246493836
 * https://github.com/coreos/fedora-coreos-tracker/issues/1297

### CODE REVIEW

**Review by bluca — APPROVED** [review]
Looks fine, but could you please add a testcase covering this in `test/test-sysusers.sh.in`? Given it's a corner case, otherwise it will risk regressing

**Comment by poettering** [comment]
hmm, can you give an example what previously wasn't allowed and now is?

File: src/sysusers/sysusers.c [hunk 1]
```diff
@@ -1175,22 +1175,32 @@ static int add_user(Item *i) {
         return 0;
 }
 
-static int gid_is_ok(gid_t gid, bool check_with_uid) {
+static int gid_is_ok(gid_t gid, char *groupname, bool check_with_uid) {
```
Comment (poettering) [inline src/sysusers/sysusers.c]: const missing

**Comment by lucab** [comment]
Rebased, const-ified, and added a new `test-sysusers/test-15` entry to cover this case.

It fixes the following scenario (observed in the wild as a somehow degenerate side-effect of nss-altfiles forwarding to initrd):
```
$ cat etc/passwd 
root:x:0:0:root:/root:/bin/bash

$ cat etc/sysusers.d/testcase.conf 
u root    0     "Super User" /root

$ systemd-sysusers --root .
Creating group 'root' with GID 999.
```

**Comment by poettering** [comment]
lgtm
